src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocTypeCheck.java

```diff
@@ -739,7 +739,6 @@ private void checkTypeParamTag(DetailAST ast,
 
         final boolean found = tags
             .stream()
-            .filter(JavadocTag::isParamTag)
```

Comment:
> We filter only params at .... so it is double check, not required.

Nope, it is not. At ` CheckUtil.getTypeParameterNames(ast); ` we only get the parameter names from the Method or class, etc. We don't check anything regarding Javadoc.

This mutation can be killed by:
```java
/**                                                                   
 *                                                                    
 * @link <T>                                                          
 */                                                                   
protected class InnerPublic2<T> // violation 'missing @param <T> tag.'
{                                                                     
}                                                                     
```
If we remove `.filter(JavadocTag::isParamTag)` then it won't give a violation which is wrong. We just compare the first argument of the tag, it may or may not be the same as the value that `@param` is supposed to have.

Reply:
Thanks a lot @Vyom-Yadav for test case